FWCore/Services/plugins/CPU.cc

```diff
@@ -125,9 +129,20 @@ namespace edm {
     }
 
 
-    void CPU::postEndJob()
+    void CPU::postEndJob() {
+        std::string models;
+        double avgSpeed;
+        Service<JobReport> reportSvc;
+        cpuInfoImpl(models, avgSpeed, &reportSvc); 
+    }
+
+    bool CPU::cpuInfo(std::string &models, double &avgSpeed)
+    {
+        return cpuInfoImpl(models, avgSpeed, nullptr);
+    }
+
+    bool CPU::cpuInfoImpl(std::string &result_models, double &result_avg_speed, Service<JobReport>* reportSvc)
```

Comment:
This seems to do a lot more than just the work need to return `result_models` and `result_avg_speed`. I think the function should be better factored so that it doesn't know about the `JobReport`.